gui/wxpython/datacatalog/tree.py

```diff
@@ -942,13 +944,18 @@ def OnDeleteMap(self, event):
 
     def OnDeleteMapset(self, event):
         """
-        Delete selected mapset
+        Delete selected mapset or mapsets
         """
         try:
-            delete_mapset_interactively(self,
-                                        self.selected_grassdb[0].data['name'],
-                                        self.selected_location[0].data['name'],
-                                        self.selected_mapset[0].data['name'])
+            db_loc_mapset = []
+            for i in range(len(self.selected_mapset)):
+                # Append to the list of tuples
+                db_loc_mapset.append(tuple(
```

Comment:
simplify the syntax to `append((db, loc, mapset))`, right now you are creating a list and converting it to tuple.

Reply:
Simplified.